### PR TITLE
feat: full `std.Address` validation

### PR DESCRIPTION
**Pull request by r3v4s**
Length of `std.Address` was only condition to check whether address is valid or not.
Fixing it by checking whether address can be decoded

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Comment by leohhhn** [comment]
Hey @r3v4s, thanks for the contribution. We initially thought about doing this, but then went with a simple length check because it would cover most human errors (ie inputting nothing for an address, or accidentally deleting a letter). I'll leave the reference for the PR that introduced the initial address validation: https://github.com/gnolang/gno/issues/1298#issuecomment-1780353231. We had an issue as the comment states, but now since #859 has been merged, your PR can work.

cc @thehowl 

Please see why the CI is failing 🙏

**Comment by r3v4s** [comment]
> Hey @r3v4s, thanks for the contribution. We initially thought about doing this, but then went with a simple length check because it would cover most human errors (ie inputting nothing for an address, or accidentally deleting a letter). I'll leave the reference for the PR/issue that introduced the initial address validation: [#1298 (comment)](https://github.com/gnolang/gno/issues/1298#issuecomment-1780353231). We had an issue as the comment states, but now since #859 has been merged, your PR can work.
> 
> cc @thehowl
> 
> Please see why the CI is failing 🙏

Hello @leohhhn, if my memory serves me this issue was covered in Miami and I forgot to tag #1298 

As you said, checking length does cover most human errors. However chain returns different result when used address has length of 40, but invalid character
> g1ubwj0apf60hd90txhnh855fkac34rxlsvua0aa

1. sending (native)ugnot using `gnokey maketx send` returns `Error: Data: invalid character not part of charset`
2. sending (realm)grc20 token using `gnokey maketx call -pkgpath {GRC20_PATH} -func Transfer -args {WRONG_FORMAT_ADDR} -args {SEND_AMOUNT}` returns non error


I think second transaction should fail too.

P.S I'll fix cause of CI Failing

**Comment by r3v4s** [comment]
@leohhhn Fixed test case causing CI to fail. Not sure If I have to assign you as reviewer or not.
> CI is still failing due to test case running too long

File: gno.land/cmd/gnoland/testdata/pr-1799.txtar
Comment (thehowl) [inline gno.land/cmd/gnoland/testdata/pr-1799.txtar:1]: please rename this to something like `grc20-invalid-address.txtar`

I've moved the other file which was named after its PR in https://github.com/gnolang/gno/pull/1810. we should not name tests after the PRs where they're introduced, but rather use a descriptive name. referencing issues makes sense only when we're fixing bad behaviour, and want to create a regression test.

File: examples/gno.land/r/demo/foo20/foo20_test.gno
Comment (thehowl) [inline examples/gno.land/r/demo/foo20/foo20_test.gno:1]: can we keep this test, but instead make unknown come from a valid (but unregistered in grc20) address?

Comment (r3v4s) [reply examples/gno.land/r/demo/foo20/foo20_test.gno:1]: I'll add address that never used in foo20(which has balance of 0), and I hope this 'fresh address' satisfy what you meant by 'ungregister in grc20'. If not, could you please provide detail conditions ?

File: gno.land/cmd/gnoland/testdata/pr-1799.txtar
Comment (r3v4s) [reply gno.land/cmd/gnoland/testdata/pr-1799.txtar:1]: done in a258319e184ed7777ff7c72a7c036a758f1e71f5

File: gnovm/stdlibs/std/crypto.gno
[the remark this answers is not in the document]
Comment (r3v4s) [reply gnovm/stdlibs/std/crypto.gno:1]: done in a6c6a49a6aee302f285fa4b0cd28156fcc723b38

File: examples/gno.land/r/demo/foo20/foo20_test.gno
Comment (r3v4s) [reply examples/gno.land/r/demo/foo20/foo20_test.gno:1]: done in 5e8e93560df0865a5be954124a82f92ede826161

**Review by petar-dambovaliev — APPROVED** [review]

**Review by thehowl — APPROVED** [review]

**Pull request merged**